extensions/positron-python/python_files/posit/positron/tests/test_variables.py

```diff
@@ -348,34 +361,41 @@ async def test_clear(
                 "version": 0,
             },
         ),
-        json_rpc_notification("refresh", {"length": 0, "variables": [], "version": 0}),
+        json_rpc_notification("refresh", {"length": 1, "variables": [underscore], "version": 0}),
     ]
 
     # All user variables are removed
     assert "x" not in shell.user_ns
     assert "y" not in shell.user_ns
 
+    # ...except hidden variables, because %reset doesn't touch those
+    assert "_" in shell.user_ns
```

Comment:
I learned that `%reset -s` (the [soft](https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-reset) reset) does not touch the underscore variable. So when we clear all variables, which calls this:

https://github.com/posit-dev/positron/blob/6e310bb5e7d9e2796f280c91000b9cbd55268724/extensions/positron-python/python_files/posit/positron/variables.py#L492

it doesn't remove the `_` from the variables pane. I decided to leave that behavior as-is instead of messing with the reset command. So now the variables pane simply reflects what's happened, in that everything gets deleted except `_`, even when it's been overridden by the user.